pkg/crc/preflight/preflight.go

```diff
@@ -0,0 +1,115 @@
+package preflight
+
+import (
+	"fmt"
+	"github.com/code-ready/crc/pkg/crc/output"
+	"github.com/spf13/viper"
+	"os"
+)
+
+type PreflightCheckFixFuncType func() (bool, error)
+
+func preflightCheckSucceedsOrFails(configNameOverrideIfSkipped string, check PreflightCheckFixFuncType, message string, configNameOverrideIfWarning string) {
+	output.Out("-- %s ... ", message)
+	isConfiguredToSkip := viper.GetBool(configNameOverrideIfSkipped)
```

Comment:
This has to be provided as a config object/struct to the checks...
... we should not have to use `viper` in the package level.

Reply:
we can hardcode this now, but have a dynamic solution later.

Reply:
you could just pass an array of `strings:bool`. you could populate this with all the viper boolean values (for now).

Reply:
Using viper here also preserves the priority of the various config values, ENV_VARIABLE vs FLAGS etc..

Reply:
let's create a new task for this.

Reply:
#28